src/cityreader/cityreader.py

```diff
@@ -14,15 +28,59 @@
 #
 # Note that the first line of the CSV is header that describes the fields--this
 # should not be loaded into a City object.
+
+# define list
 cities = []
 
 def cityreader(cities=[]):
-  # TODO Implement the functionality to read from the 'cities.csv' file
-  # Ensure that the lat and lon valuse are all floats
-  # For each city record, create a new City instance and add it to the 
-  # `cities` list
+  """
+  read from the 'cities.csv' file; 
+  each record should be a City instance;
+  parse header from CSV on import;
+
+  return a list with all City instances
+  """
+  # Open the csv file.
+  # with open('cities.csv', 'r') as my_file:
+    # Create a csv.reader.
+    # reader = csv.reader(my_file)
+  
+  file_path = 'src/cityreader/cities.csv'
+  df = pd.read_csv(file_path)
+
+  dict_list = []
+
+  # iterate over each row in df
+  for i in df.index:
+    # create empty dictionary for each row
+    dict_row = {}
+    # populate 
+    dict_row['name'] = df.at[i, 'city']
+    dict_row['lat'] = df.at[i, 'lat']
+    dict_row['lon'] = df.at[i, 'lng']
+
+    # append
+    dict_list.append(dict_row)
+    # print(dict_list)
+  for i in dict_list:
+    print(i)
+    new_city = City(i['name'], float(i['lat']),float(i['lon']))
+    print(new_city)
+    cities.append(new_city)
+    print(cities)
```

Comment:
This works well to show the steps taken to achieve the MVP result. The final code might be good commenting out the extra prints, but all good so far.